src/anemoi/inference/runner.py

```diff
@@ -226,6 +226,11 @@ def run(
         if start_datetime is None:
             start_datetime = input_fields.order_by(valid_datetime="ascending")[-1].datetime()
 
+        if isinstance(start_datetime, dict):
+            start_datetime = start_datetime[
+                "base_time"
+            ]  # With earthkit, time is a dictionary with 'base_time' and 'valid_time'
```

Comment:
Can you put the comment on a new line so the linter doesn't split the assignment into multiple lines. You could put it above the assignment or above the if.

Reply:
Resolved in upcoming commit

Reply:
Can you use:

`start_datetime = x.order_by(valid_datetime="ascending")[-1].metadata('valid_datetime')`